csrc/python_frontend/fusion_definition.cpp

```diff
@@ -344,7 +347,7 @@ void FusionDefinition::print(std::ostream& os) const {
   os << std::endl;
 }
 
-std::vector<at::Tensor> FusionDefinition::execute(
+std::vector<DistributedTensor> FusionDefinition::execute(
```

Comment:
Alternatives: 
1. Return `std::vector<std::variant<at::Tensor, DistributedTensor>>`. Because `DistributedTensor` can also represent a non-distributed tensor, I chose the current API for simplicity -- C++ is more verbose than Python when dealing with dynamic types. 
2. Return `std::variant<std::vector<at::Tensor>, std::vector<DistributedTensor>>`. Same reason. 
3. Store output shardings (i.e. the mesh and the mesh-to-tenseor-axis mapping) to a field of FusionDefinition and retrieve it using another method. This would be similar to getDebugOutput. I didn't choose that because it introduced a new state in the class that could get out of sync.